code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -34,94 +34,501 @@
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			var/obj/item/stack/sheet/glass/O = new(null, 5)
+			if(!user.put_in_active_hand(O))
+				O.forceMove(user.drop_location())
+			playsound(user, 'sound/effects/splat.ogg', 50, 1)
+			user.visible_message("<span class='warning'>[user] spits out [O]!</span>", "<span class='notice'>You spit out [O]!</span>")
+			return 150
+
+		if(SLIME_ACTIVATE_MAJOR)
+			var/obj/item/stack/sheet/metal/O = new(null, 5)
+			if(!user.put_in_active_hand(O))
+				O.forceMove(user.drop_location())
+			playsound(user, 'sound/effects/splat.ogg', 50, 1)
+			user.visible_message("<span class='warning'>[user] spits out [O]!</span>", "<span class='notice'>You spit out [O]!</span>")
+			return 200
+
 /obj/item/slime_extract/purple
 	name = "purple slime extract"
 	icon_state = "purple slime extract"
 
+/obj/item/slime_extract/purple/activate(mob/living/carbon/human/user, datum/species/jelly/luminescent/species, activation_type)
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			user.nutrition += 50
+			user.blood_volume += 50
+			to_chat(user, "<span class='notice'>You activate [src], and your body is refilled with fresh slime jelly!</span>")
+			return 150
+
+		if(SLIME_ACTIVATE_MAJOR)
+			to_chat(user, "<span class='notice'>You activate [src], and it releases regenerative chemicals!</span>")
+			user.reagents.add_reagent("tricordrazine",10)
+			return 600
+
 /obj/item/slime_extract/darkpurple
 	name = "dark purple slime extract"
 	icon_state = "dark purple slime extract"
 
+/obj/item/slime_extract/darkpurple/activate(mob/living/carbon/human/user, datum/species/jelly/luminescent/species, activation_type)
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			var/obj/item/stack/sheet/mineral/plasma/O = new(null, 1)
+			if(!user.put_in_active_hand(O))
+				O.forceMove(user.drop_location())
+			playsound(user, 'sound/effects/splat.ogg', 50, 1)
+			user.visible_message("<span class='warning'>[user] spits out [O]!</span>", "<span class='notice'>You spit out [O]!</span>")
+			return 150
+
+		if(SLIME_ACTIVATE_MAJOR)
+			var/turf/open/T = get_turf(user)
+			if(istype(T))
+				T.atmos_spawn_air("plasma=20")
+			to_chat(user, "<span class='warning'>You activate [src], and a cloud of plasma bursts out of your skin!</span>")
+			return 900
+
 /obj/item/slime_extract/orange
 	name = "orange slime extract"
 	icon_state = "orange slime extract"
 
+/obj/item/slime_extract/orange/activate(mob/living/carbon/human/user, datum/species/jelly/luminescent/species, activation_type)
+	switch(activation_type)
+		if(SLIME_ACTIVATE_MINOR)
+			to_chat(user, "<span class='notice'>You activate [src]. You start feeling hot!</span>")
+			user.reagents.add_reagent("capsaicin",10)
+			return 150
+
+		if(SLIME_ACTIVATE_MAJOR)
+			user.reagents.add_reagent("phosphorus",5)//
+			user.reagents.add_reagent("potassium",5) // = smoke, along with any reagents inside mr. slime
```

Comment:
That additional comment `//smoke, along with any reagents inside mr. slime` is worrying, because it implies you could nuke people silently with toxic substances the slimes can heal from and not die in a sudden smoke burst of amatoxin (which is pure toxic and has no additional effects) that'll stack especially in a public place if you bit a dental implant or chemical implant.

Reply:
Fun fact, you can already do this with a beaker and a classic slimeperson

Reply:
The point is that you can chug a huge amount of chemical reagents contained within yourself and release it in the ability, the largest comparable container to stabilize & contain the smoke plus reagent would be a bluespace beaker.

Reply:
You can drink a poison cocktail up to 985 u then sip the ingredients for smoke.

Reply:
I was going to insert a meme picture about a certain Croatian general but even my taste isn't that poor, but surely you can see that if its not instantly transferred into the smoke you're going to still have it in your system metabolising slowly for the next cooldown.

Reply:
it's instantly transferred into the smoke

the old 'mix smoke in a poisoned person to de-poison them' has existed forever

Reply:
WHAT IS THIS TECH WTFFFFF